libeos-updater-util/avahi-service-file.c

```diff
@@ -257,3 +326,426 @@ eos_avahi_service_file_delete (const gchar   *avahi_service_directory,
+  if (!get_summary_timestamp (repo, cancellable, &summary_timestamp, error))
+    {
+      return FALSE;
+    }
+  summary_timestamp_variant = variant_sink_new ("t", summary_timestamp);
+  g_assert (g_variant_type_equal (g_variant_get_type (summary_timestamp_variant),
+                                  EOS_OSTREE_AVAHI_V1_SUMMARY_TIMESTAMP_VARIANT_TYPE));
+  repository_index_variant = variant_sink_new ("q", repository_index);
+  g_assert (g_variant_type_equal (g_variant_get_type (repository_index_variant),
+                                  EOS_OSTREE_AVAHI_V1_REPOSITORY_INDEX_VARIANT_TYPE));
+
+  service_file = get_ostree_service_file (avahi_service_directory,
+                                          repository_index);
+
+  return generate_ostree_avahi_v1_service_file_from_variants (service_file,
+                                                              port,
+                                                              version_variant,
+                                                              refs_bloom_filter_variant,
+                                                              summary_timestamp_variant,
+                                                              repository_index_variant,
+                                                              cancellable,
+                                                              error);
+}
+
+gboolean
+eos_ostree_avahi_service_file_generate (const gchar   *avahi_service_directory,
+                                        OstreeRepo    *repo,
+                                        GVariant      *options,
+                                        GCancellable  *cancellable,
+                                        GError       **error)
+{
+  g_auto(GVariantDict) options_dict = G_VARIANT_DICT_INIT (options);
+  guchar version = 1;
+
+  g_return_val_if_fail (avahi_service_directory != NULL, FALSE);
+  g_return_val_if_fail (OSTREE_IS_REPO (repo), FALSE);
+  g_return_val_if_fail (options == NULL ||
+                        g_variant_type_equal (g_variant_get_type (options),
+                                              G_VARIANT_TYPE_VARDICT),
+                        FALSE);
+  g_return_val_if_fail (cancellable == NULL || G_IS_CANCELLABLE (cancellable), FALSE);
+  g_return_val_if_fail (error == NULL || *error == NULL, FALSE);
+
+  g_variant_dict_lookup (&options_dict, EOS_OSTREE_AVAHI_OPTION_FORCE_VERSION_Y,
+                         "y", &version);
+  switch (version)
+    {
+    case 1:
+      return generate_ostree_avahi_v1_service_file (avahi_service_directory,
+                                                    repo,
+                                                    &options_dict,
+                                                    cancellable,
+                                                    error);
+
+    default:
+      break;
+    }
+
+  g_set_error (error, G_IO_ERROR, G_IO_ERROR_FAILED,
```

Comment:
The other case handling does it in a different way only because there was some more code after `switch`. I can move this inside the default - I hope that the compiler won't start complaining about missing `return`.